### PR TITLE
Feature/badges

### PR DESCRIPTION
**Pull request by KianMiridoozini**
feature/badges: 

Users now have badges.
Badges are based on the number of shared recipes.
The badge is currently only displayed on the user's profile page.

Users will get notified after completion of sharing the cocktail that fulfills the count-criteria for the respective badge.
A progress-bar has been added to user profile.

A progress-bar has been implemented for the badges.
This progress-bar displayes and indicates the progress left until next badge.

##Future Implementations:
- Addition of Images/icons for each badge
- Badge icon display on control panel
- Badge icon display next to username

##Known issues:
- Image upload js functionalities.
- Newly created accounts see error on control panel. This issue has to do with lack of info. The error goes away after providing profile information like first_name, last_name, and profile_picture.

### CODE REVIEW

File: app/helpers/helpers.php [hunk 1]
```diff
@@ -54,9 +54,6 @@ function sanitizeNumber($input) {
     }
     return null; 
 }
-function sanitizeTrim($input) {
-    return is_string($input) ? trim($input) : $input;
-}
```
Comment (jayb2302) [inline app/helpers/helpers.php:57]: Why did you remove sanitizeTrim ?

Comment (KianMiridoozini) [reply app/helpers/helpers.php:57]: After pulling from develop, i got duplicate of that function. That function still exists in the helpers.

File: app/views/cocktails/view.php [hunk 2]
```diff
@@ -16,6 +16,14 @@
...
 $cocktailTitle = htmlspecialchars($cocktail->getTitle() ?? 'Unknown Cocktail');
 ?>
+<?php if (isset($_SESSION['badge_notification'])): ?>
+    <?php error_log("Displaying badge notification: " . print_r($_SESSION['badge_notification'], true)); ?>
+    <div class="alert alert-success">
+        <h4><?= htmlspecialchars($_SESSION['badge_notification']['message']); ?></h4>
+        <p><?= htmlspecialchars($_SESSION['badge_notification']['badge_description']); ?></p>
+    </div>
+    <?php unset($_SESSION['badge_notification']); ?>
+<?php endif; ?>
```
Comment (jayb2302) [inline app/views/cocktails/view.php:26]: IS there a reason why this is in cocktails/view.php ?

Comment (KianMiridoozini) [reply app/views/cocktails/view.php:26]: After the users share a recipe, they are directed to the recipe they have shared.
It made most sense to display the notification right after completion of their milestone-recipe on the cocktail page.

**Review by jayb2302 — APPROVED** [review]
Awesome, very nice work 🔥
